### PR TITLE
Add code coverage to js/noms

### PR DESCRIPTION
**Pull request by arv**

### CODE REVIEW

File: js/noms/package.json [hunk 1]
```diff
@@ -37,12 +37,14 @@
     "documentation": "documentationjs/documentation#a71bb03681cd145723218910145c835dd5171a74",
     "flow-bin": "0.30.0",
     "fs-extra": "^0.30.0",
-    "mocha": "^2.5.3"
+    "mocha": "^2.5.3",
+    "nyc": "^8.3.0"
```
Comment (mikegray) [inline js/noms/package.json]: Do we need to also add this to js/perf/*? and samples/js/*?

File: js/noms/package.json [hunk 2]
```diff
@@ -37,12 +37,14 @@
...
   },
   "scripts": {
     "pretest": "eslint src/ && flow src/",
     "ts": "mocha --ui tdd --timeout=4000 --reporter spec --slow=50 --invert --grep='LONG:' --compilers js:babel-core/register src/*-test.js",
-    "test": "mocha --ui tdd --timeout=6000 --reporter dot --compilers js:babel-core/register src/*-test.js",
+    "test": "mocha --ui tdd --timeout=10000 --reporter dot --compilers js:babel-core/register src/*-test.js",
+    "code-cov": "nyc npm test && nyc report --reporter=text-lcov > coverage.lcov",
```
Comment (mikegray) [inline js/noms/package.json]: Hmm, it must be expensive - is that why you are not running it whenever tests run? Also doesn't make sense to run it whenever any random person runs `npm test` probably. Maybe we just need a CI specific target?

Comment (arv) [inline js/noms/package.json]: Yeah maybe we should rerun the tests?

File: js/noms/package.json (hunk 1 above)
Comment (arv) [inline js/noms/package.json]: Not at this time.

File: js/noms/package.json (hunk 2 above)
Comment (arv) [inline js/noms/package.json]: Actually, we already have `npm run ts` which skips long tests. I'll just make `npm test` run the code coverage.

**Comment by mikegray** [comment]
lgtm

**Comment by arv** [comment]
PTAL

**Pull request merged**